TileStache/Goodies/Providers/MapnikGrid.py

```diff
@@ -125,17 +125,17 @@ def save(self, out, format):
 
         out.write(self.content)
 
-def get_mapnikMap(mapfile):
-    """ Get a new mapnik.Map instance for a mapfile
+def get_mapnikMap(mapconfig):
+    """ Get a new mapnik.Map instance for a mapconfig
     """
     mmap = mapnik.Map(0, 0)
 
-    if exists(mapfile):
-        mapnik.load_map(mmap, str(mapfile))
+    if exists(mapconfig):
```

Comment:
This will fail if you pass an XML string, wouldn't it? And then it will try to do the lines below.